### ISSUE TITLE
Swimming Pools and Water Handling Unit 

### ISSUE THREAD

**Issue by annaPoensgen** [RWTH-EBC/AixLib#1071]
**Context**

This issue is part of the planned TEASER extension, which adds the new building archetype  swimming bath. The extension is needed to depict the overall energy and water demands of a swimming bath. 

A swimming bath building contains multiple zones, which can contain one or more swimming pools, e. g. swimming hall and Spa/Sauna Area.

Goal of implementing swimming pools in those zones is to calculate the water and energy demand of the swimming pool as accurate as possible. This includes the correct modeling of the exchange between the swimming pool and its surroundings. 

All swimming pools are connected to an individual water treatment unit, where heat and water losses of the water cylce are compensated. 

Heat losses include:
- Convection interactions with the surrounding air
- Evaporation losses, due to water evaporating on the pool surface
- Conduction through the swimming pool walls
- Radiation exchanges between the water surface and the surrounding walls 
- Exchange of warm circuit water with cold fresh water (due to filter cleansing and to thin out non filterable solvents);  next to cold fresh water the added fresh water can be warm recycled water.

**What is the problem?**

Currently there is no fitting model in the AixLib, which could be used to approximate the water and energy demands of swimming pools sufficiently.  The before mentioned processes are not included as we would need them in the existing tank or storage models. 

The moisture intake of the air in zones with swimming pools is way higher than in zones without swimming pools. The evaporation rate from swimming pools is depending on the current humidity and temperature of the surrounding zone air. Thus the calculation of the actual moisture gains has to be simultaneously to the ongoing heat and air exchange simulation in the zone and between the zone and ventilation system.  

In the currently used model ThermalZoneMoistAirExchange the implemented moistureGains block feeds a constant pre-calculated value of moisture gains to the room air and is therefore not fitting to describe the fluctuating moisture gains due to swimming pools. 

**How do we want to solve it?**

The current idea is to implement a swimming pool model (including all the necessary parts as the water volume, swimming pool walls, port for evaporating water etc. ) and to add the according number of pool models into the ThermalZoneMoistAirExchange model.  

We want to include the water treatment processes by adding a Water Handling Unit (WHU) into the MoistAirEquipped model, based on the model of the Air Handling Unit. The WHU then includes one water treatment model for each swimming pool and additionally two shared water sources (warm and cold) and one water sink. For now, we don’t want to distinguish between sinks for recycled water and sewerage water, but that is not finally decided yet.  

Therefore we will have to create alternative models for both, the ThermalZoneMoistAirExchange and the MoistAirEquipped model, to include the planned WHU and Swimming Pool models.  

**Describe alternatives we've considered**

We considered two other alternatives of integrating the swimming pool model:
1. Implementing the Swimming Pools into the ROM Four Elements
2. Implementing the Swimming Pools each as their own zone

We are not further considering the both alternatives, due to the need of ROM alteration, which is not desirable as it is not part of the AixLib but the IBPSA.

The WHU could consist of one Water-Treatment-Unit and one Sources-and-Sink model. The individual in- and outgoing mass flows could be realized with vectors. The individual heat demands of each pool could still be calculated. The design process of the WHU is not finished and still ongoing.  

**Additional context**

The two added pictures are drafts of the initial idea on how two implement the Swimming Pool and the Water Treatment (Handling) Unit.
 
![ThermalMoistAirExchange](https://user-images.githubusercontent.com/72392836/106127685-09d9fb00-615f-11eb-8ad6-1421cd1ec3a6.jpg)

![MoistAirEquipped](https://user-images.githubusercontent.com/72392836/106127690-0b0b2800-615f-11eb-8dd6-251a804e0a6e.jpg)

**Comment by LauraMaier** [comment]
@annaPoensgen and @larissakuehn : I suppose this issue is closed within the next weeks/months once your master thesis is done? Of what is the current status?
Remember to link the issue in your commit message with #ISSUENo so that we can follow your updates :-)

**Comment by FelixStege** [comment]
@larissakuehn, what is the current status here? Maybe, this is something for the upcoming hackday?

**Comment by larissakuehn** [comment]
The issue is no longer relevant and has been replaced by issue 1407 and issue 1417.
